backend/pkg/web/handler/access_token.go

```diff
@@ -0,0 +1,169 @@
+package handler
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/fastenhealth/fasten-onprem/backend/pkg"
+	"github.com/fastenhealth/fasten-onprem/backend/pkg/auth"
+	"github.com/fastenhealth/fasten-onprem/backend/pkg/config"
+	"github.com/fastenhealth/fasten-onprem/backend/pkg/database"
+	"github.com/fastenhealth/fasten-onprem/backend/pkg/models"
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+// InitiateAccess generates a new access token for mobile app authentication
+type InitiateAccessRequest struct {
+	Name       string `json:"name"`
+	Expiration int    `json:"expiration"` // Expiration in days. 0 for no expiration.
+}
+
+// CreateAccessToken generates a new access token for mobile app authentication
+func CreateAccessToken(c *gin.Context) {
+	log := c.MustGet(pkg.ContextKeyTypeLogger).(*logrus.Entry)
+	databaseRepo := c.MustGet(pkg.ContextKeyTypeDatabase).(database.DatabaseRepository)
+	appConfig := c.MustGet(pkg.ContextKeyTypeConfig).(config.Interface)
+
+	var req InitiateAccessRequest
+	if err := c.ShouldBindJSON(&req); err != nil {
+		// If there's an error binding, or no name is provided, proceed without a specific name
+		log.Debugf("No device name provided or invalid request format: %v", err)
+	}
+
+	log.Debug("Attempting to get current user for access initiation")
```

Comment:
most of this logic should be in the Database repository, its the security boundary for the app.